src/components/ha-faded.ts

```diff
@@ -0,0 +1,75 @@
+import { css, CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+
+@customElement("ha-faded")
+class HaFaded extends LitElement {
+  @property({ type: Number }) public fadedHeight = 102;
+
+  @state() _contentShown = false;
+
+  @query(".container") private _container!: HTMLDivElement;
+
+  protected render(): TemplateResult {
+    return html`
+      <div
+        class="container ${classMap({ faded: !this._contentShown })}"
+        style=${!this._contentShown ? `max-height: ${this.fadedHeight}px` : ""}
+        @click=${this._showContent}
+      >
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  get _slottedHeight(): number {
+    const elements = (
+      this._container.firstElementChild as HTMLSlotElement
+    ).assignedElements();
+
+    if (!elements.length) {
+      return this._container.clientHeight;
+    }
+    return elements.reduce(
+      (partial, element) => partial + element.clientHeight,
+      0
+    );
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    if (this._slottedHeight < this.fadedHeight) {
```

Comment:
Can we add a small buffer? Like, if clicking on the "fade" shows just 2 more lines, we should have just shown those 2 lines instead of fading.

Reply:
What are 2 lines?
16px?
2xh1?

Reply:
50px ?